rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/IFeaturesFactory.kt

```diff
@@ -1,41 +1,100 @@
+/*
+ * Copyright 2019 Sme.UP S.p.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 @file:JvmName("StandardFeaturesFactory")
 package com.smeup.rpgparser.interpreter
 
 import java.lang.IllegalArgumentException
 import kotlin.reflect.full.createInstance
+import com.smeup.rpgparser.parsing.parsetreetoast.RpgType
 
 /**
- * Allows enable features
+ * Allows to enable features
  * */
 interface IFeaturesFactory {
     fun createSymbolTable(): ISymbolTable
+
+    /**
+     * It allows to override the StringType creation.
+     * The current implementation tests the presence of [FeatureFlag.UnlimitedStringTypeFlag] and
+     * if is set to on it returns an instance of [UnlimitedStringType]
+     * @param create: default creation type implementation
+     * @return the instance of type created
+     * */
+    fun createStringType(create: () -> StringType): Type {
+        return if (FeatureFlag.UnlimitedStringTypeFlag.isOn()) {
+            UnlimitedStringType
+        } else create.invoke()
+    }
 }
 
 object FeaturesFactory {
 
     private val factory: IFeaturesFactory by lazy {
-        val property = System.getProperty("featuresFactory", "")
+        val property = System.getProperty("jariko.featuresFactory", System.getProperty("featuresFactory", ""))
         val featuresFactoryId = if (property == "") "default" else {
             property
         }
         val featuresFactoryImpl = if (featuresFactoryId.contains('.', false)) {
             featuresFactoryId
         } else {
-            val property = java.util.Properties()
+            val mProperty = java.util.Properties()
```

Comment:
Why calls mProperty ? what means "m"?

Reply:
I changed the name because the variable name `property` shadowed a variable named in the same way defined at the method beginning.